File: ComponentImpl/src/main/java/com/xiaojinzi/component/bean/ActivityResult.java
package com.xiaojinzi.component.bean;

import android.content.Intent;
import android.support.annotation.Nullable;

import com.xiaojinzi.component.error.ignore.ActivityResultException;

/**
 * activity result 的返回对象,{@link android.app.Activity#onActivityResult(int, int, Intent)}
 * time   : 2018/12/04
 *
 * @author : xiaojinzi 30212
 */
public class ActivityResult {

    public final int requestCode;

    public final int resultCode;

    @Nullable
    public final Intent data;

    public ActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        this.requestCode = requestCode;
        this.resultCode = resultCode;
        this.data = data;
    }

    public Intent intentCheckAndGet() throws ActivityResultException {
        if (data == null) {
            throw new ActivityResultException("the intent result data is null");
        }
        return data;
    }

    public Intent intentWithResultCodeCheckAndGet(int expectedResultCode) {
        if (data == null) {
            throw new ActivityResultException("the intent result data is null");
        }
        if (expectedResultCode != resultCode) {
            throw new ActivityResultException("the resultCode is not matching " + expectedResultCode);
        }
        return data;
    }

}

File: ComponentImpl/src/main/java/com/xiaojinzi/component/bean/ActivityResult.kt
package com.xiaojinzi.component.bean

import android.content.Intent
import com.xiaojinzi.component.error.ignore.ActivityResultException

/**
 * activity result 的返回对象,[android.app.Activity.onActivityResult]
 * time   : 2018/12/04
 *
 * @author : xiaojinzi
 */
data class
ActivityResult(
        val requestCode: Int,
        val resultCode: Int,
        val data: Intent?
) {

    @Throws(ActivityResultException::class)
    fun intentCheckAndGet(): Intent {
        if (data == null) {
            throw ActivityResultException("the intent result data is null")
        }
        return data
    }

    @Throws(ActivityResultException::class)
    fun intentWithResultCodeCheckAndGet(expectedResultCode: Int): Intent {
        if (data == null) {
            throw ActivityResultException("the intent result data is null")
        }
        if (expectedResultCode != resultCode) {
            throw ActivityResultException("the resultCode is not matching $expectedResultCode")
        }
        return data
    }

}